### ISSUE TITLE
Spacing within heading blocks

### ISSUE THREAD

**Issue by natrow** [typst/typst#2351]
### Description

It would be helpful if we could horizontally space/padding to the header title to separate it from the numbering, similar to below:

![image](https://github.com/typst/typst/assets/60853993/12a5630f-c21e-45a3-97ce-27abeba62f0c)

### Use Case

Visual padding between the title and numbering can make the text more readable and may be required by some formats.

**Comment by laurmaedje** [comment]
You can do this with a show rule, but I agree that it could be a bit easier.
```typ
#set heading(numbering: "1.1")
#show heading: it => block({
  box(width: 35pt, counter(heading).display())
  it.body
})

= Introduction
#lorem(40)

== Content
#lorem(40)

=== In This Report
#lorem(40)
```
![image](https://github.com/typst/typst/assets/17899797/8e512ae2-b36e-4762-bd8f-095241349f67)

**Comment by natrow** [comment]
Ah makes sense, I enjoy Typst so far but learning all of these small tricks with the formatting I want isn't always intuitive. I think this would be a good example in the heading documentation because this is probably a fairly common use-case. Or even as a setting in the web GUI.

**Comment by natrow** [comment]
Edit: there is one problem with that function
![image](https://github.com/typst/typst/assets/60853993/0647cfd8-8339-495b-9392-84f4821b6a02)

Headers that aren't numbered shouldn't have the function applied.

A corrected version:

```rs
show heading: it => if it.numbering != none {
    block({
      box(width: 0.5in, counter(heading).display())
      it.body
    })
  } else {
    it
  }
```

**Comment by laurmaedje** [comment]
Yeah, you can also add an `if it.numbering != none` around the counter.
